Casks/n/nao.rb

```diff
@@ -0,0 +1,38 @@
+cask "nao" do
+  arch arm: "arm64", intel: "x64"
+
+  on_arm do
+    version "0.1.27"
+    sha256 "fa95235139be9b4e372eef9e24fd0f76056995a8ddc4deaff7bac9e76586fe9a"
+  end
+  on_intel do
+    version "0.1.26"
+    sha256 "ac5b136ad29331f565da2e2ce68e030f7ea862528fb2540d0138ef7e80a31162"
+  end
+
+  url "https://storage.googleapis.com/nao-releases/nao/darwin-#{arch}/#{version}/nao.#{arch}.#{version}.dmg",
+      verified: "storage.googleapis.com/nao-releases/nao/"
+  name "nao"
+  desc "AI code editor for data"
+  homepage "https://getnao.io/"
+
+  livecheck do
+    url "https://storage.googleapis.com/nao-releases/nao/darwin-#{arch}/releases.json"
+    strategy :json do |json|
+      json["currentRelease"]
+    end
+  end
+
+  depends_on macos: ">= :catalina"
+
+  app "nao.app"
```

Comment:
```suggestion
  app "nao.app"
  binary "#{appdir}/nao.app/Contents/Resources/app/bin/nao"
```

We can probably add a binary since it's a VS Code fork.